server/src/test/java/org/opensearch/cluster/coordination/CoordinationStateTests.java

```diff
@@ -1002,6 +1001,168 @@ public void testHandlePrePublishAndCommitWhenRemoteStateEnabled() throws IOExcep
+                "",
+                REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY,
+                ""
+            )
+        );
+        // cluster manager is node1 and node2 is a follower node
+        VotingConfiguration initialConfig = VotingConfiguration.of(node1);
+        ClusterState state1 = clusterState(
+            0L,
+            0L,
+            DiscoveryNodes.builder()
+                .add(node1)
+                .add(nodeWithPub)
+                .clusterManagerNodeId(node1.getId())
+                .localNodeId(nodeWithPub.getId())
+                .build(),
+            initialConfig,
+            initialConfig,
+            42L
+        );
+        final PersistedStateRegistry persistedStateRegistry = persistedStateRegistry();
+        persistedStateRegistry.addPersistedState(PersistedStateType.LOCAL, new InMemoryPersistedState(0L, initialStateNode2));
+        persistedStateRegistry.addPersistedState(
+            PersistedStateType.REMOTE,
+            new RemotePersistedState(remoteClusterStateService, state1.metadata().clusterUUID())
+        );
+
+        final CoordinationState coordinationState = createCoordinationState(
+            persistedStateRegistry,
+            nodeWithPub,
+            remotePublicationSettings()
+        );
+        coordinationState.setInitialState(state1);
+        long newTerm = randomLongBetween(1, 10);
+        StartJoinRequest startJoinRequest = new StartJoinRequest(nodeWithPub, newTerm);
+
+        coordinationState.handleStartJoin(startJoinRequest);
+
+        ClusterState state2 = setValue(
+            ClusterState.builder(state1)
+                .metadata(
+                    Metadata.builder(state1.metadata())
+                        .coordinationMetadata(CoordinationMetadata.builder(state1.coordinationMetadata()).term(newTerm).build())
+                        .build()
+                )
+                .version(randomLongBetween(1, 10))
+                .build(),
+            43L
+        );
+        final ClusterMetadataManifest manifest = ClusterMetadataManifest.builder()
+            .clusterTerm(1L)
+            .stateVersion(state2.version())
+            .clusterUUID(state2.metadata().clusterUUID())
+            .nodeId(node1.getId())
+            .stateUUID(randomAlphaOfLength(10))
+            .opensearchVersion(Version.CURRENT)
+            .committed(false)
+            .codecVersion(1)
+            .globalMetadataFileName(randomAlphaOfLength(10))
+            .indices(Collections.emptyList())
+            .previousClusterUUID(randomAlphaOfLength(10))
+            .clusterUUIDCommitted(true)
+            .build();
+
+        PublishResponse publishResponse = coordinationState.handlePublishRequest(new PublishRequest(state2, manifest));
+        assertEquals(state2.term(), publishResponse.getTerm());
+        assertEquals(state2.version(), publishResponse.getVersion());
+        verifyNoInteractions(remoteClusterStateService);
```

Comment:
Wont the follower needs need to read updated coordinationMetadata?

Reply:
This is not required in this unit of the code. As we would have already downloaded the updated metadata and created the updated state and we are providing that to this method to handle that updated state.